packages/pyolite-kernel/py/pyolite/pyolite/kernel.py

```diff
@@ -1,14 +1,17 @@
 # This is our ipykernel mock
 from ipykernel import CommManager
 
-from .display import display_publisher
+from .custom_display import XDisplayHook, XDisplayPublisher
 from .interpreter import Interpreter
-from .custom_display import XDisplayHook
+
 
 class Pyolite:
     def __init__(self):
-        self.interpreter = Interpreter(kernel=self, displayhook_class=XDisplayHook)
-        self.display_publisher = display_publisher
```

Comment:
And `self.display_publisher` would be `self.interpreter.display_pub`

Reply:
we won't need to use it anymore, because we are not doing `kernel.display_publisher.display_callback = displayCallback;`

we are directly doing

```
  interpreter.display_pub.clear_output_callback = clearOutputCallback;
  interpreter.display_pub.display_data_callback = displayDataCallback;
  interpreter.display_pub.update_display_data_callback = updateDisplayDataCallback;
```

Reply:
Marking as resolved.